src/transport.h

```diff
@@ -34,13 +34,14 @@ class Transport : public AttributeConverter<RequestType> {
   // Send the attributes
   void Send(const Attributes& attributes, ResponseType* response,
             DoneFunc on_done) {
+    std::lock_guard<std::mutex> lock(mutex_);
     stream_.Call(attributes, response, on_done);
   }
 
   // Convert to a protobuf
+  // This is called by stream_.Call so it is within the mutex lock.
   void FillProto(StreamID stream_id, const Attributes& attributes,
```

Comment:
Should FillProto be a private method then, considering it needs to be called with the private lock being held?